tests/devices/fetch_network.js

```diff
@@ -202,19 +250,61 @@ const tests =
     },
 ];
 
-const emulator = new V86({
-    bios: { url: __dirname + "/../../bios/seabios.bin" },
-    vga_bios: { url: __dirname + "/../../bios/vgabios.bin" },
-    bzimage: { url: __dirname + "/../../images/buildroot-bzimage68.bin" },
-    autostart: true,
-    memory_size: 64 * 1024 * 1024,
-    disable_jit: +process.env.DISABLE_JIT,
-    net_device: {
-        relay_url: "fetch",
-        type: USE_VIRTIO ? "virtio" : "ne2k",
-    },
-    log_level: SHOW_LOGS ? 0x400000 : 0,
-});
+if(isMainThread)
+{
+    emulator = new V86({
+        bios: { url: __dirname + "/../../bios/seabios.bin" },
+        vga_bios: { url: __dirname + "/../../bios/vgabios.bin" },
+        bzimage: { url: __dirname + "/../../images/buildroot-bzimage68.bin" },
+        autostart: true,
+        memory_size: 64 * 1024 * 1024,
+        disable_jit: +process.env.DISABLE_JIT,
+        net_device: {
+            relay_url: "fetch",
+            type: USE_VIRTIO ? "virtio" : "ne2k",
+            local_http: true
+        },
+        log_level: SHOW_LOGS ? 0x400000 : 0,
+    });
+
+    server = new Worker(__filename);
+    server.on("error", (e) => { throw new Error("server: " + e); });
+    server.on("message", function(msg) {
+        SERVER_PORT = msg;
+        console.log("Server started on port " + SERVER_PORT);
+    });
+}
+else
+{
+    // placeholder for main thread handlers
+    emulator = { bus: {} };
+    emulator.add_listener = emulator.bus.register = () => null;
```

Comment:
Wouldn't it be simpler to move the calls to emulator.add_listener etc. into the if branch above and delete this code?